index.js

```diff
@@ -26,6 +26,18 @@ if (!isRegexSafe(FULL_PATH_REGEXP)) {
 
 const acceptVersionStrategy = require('./lib/accept-version')
 
+function buildMethodMap () {
+  const code = []
+  for (var i = 0; i < http.METHODS.length; i++) {
+    var m = http.METHODS[i]
+    code.push(`this['${m}'] = null`)
+  }
+  return new Function(code.join('\n')) // eslint-disable-line
+}
```

Comment:
Why the use of new Function here?

Reply:
I was just copying the code that was already there -- this object with one property per HTTP method was used in `Node` for the handler map before. See https://github.com/delvedor/find-my-way/commit/c70b2e8599262aa4d784f8f50c88ee655cca94ca. I think it a little unclear and since there's only one instance of them per router I will switch it back to being a plain old constructor.

Reply:
I take it back -- this makes a huge difference:

```

function buildMethodMap () {
  const code = []
  for (var i = 0; i < http.METHODS.length; i++) {
    var m = http.METHODS[i]
    code.push(`this['${m}'] = null`)
  }
  return new Function(code.join('\n')) // eslint-disable-line
}

// Object with prototype slots for all the HTTP methods
const MethodMap = buildMethodMap()
```

gives me `lookup static route x 44,563,670 ops/sec ±0.88% (584 runs sampled)`

whereas a constructor that just loops over the methods gives me `lookup static route x 29,380,113 ops/sec ±1.03% (573 runs sampled)`